### PR TITLE
Refactor PDF command classes.

### PR DESCRIPTION
**Pull request by tmlippman**
Replace namedtuple inheritance with metaclasses and add a COMMAND
class attribute to all PDF commands. We'll use this to implement
content-stream parsing: taking in a string of PDF commands and
returning a list of PDF command instances.

### CODE REVIEW

**Comment by tmlippman** [comment]
@mjbryant the parsing stuff isn't here yet, clearly.

File: pdf_annotate/graphics.py [hunk 1]
```diff
@@ -70,135 +78,244 @@ def join(stream1, stream2):
         return ContentStream(stream1.commands + stream2.commands)
 
 
-class NoOpTransformBase(object):
+@total_ordering
```
Comment (mjbryant) [inline pdf_annotate/graphics.py]: Why are you using `@total_ordering` if you're just raising an error in `__lt__`? You're not defining comparisons in any other subclasses.

Comment (tmlippman) [reply pdf_annotate/graphics.py]: `@total_ordering` propagates the error into `__gt__`, `__gte__`, etc. I wanted the PDF command classes to be un-orderable, so you can't sort them etc. This prevents anyone relying on the tuple ordering that many of these classes have.

Comment (mjbryant) [reply pdf_annotate/graphics.py]: Ohhh, so you're using this to force them to _not_ be orderable. Sneaky.

**Pull request merged**